modules/internal/OwnedObject.chpl

```diff
@@ -245,16 +245,37 @@ module OwnedObject {
        refer to `nil` after this call.
      */
     proc init=(pragma "leaves arg nil" pragma "nil from arg" ref src:_owned) {
+      if isNonNilableClass(this.type) && isNilableClass(src) &&
+         !chpl_legacyNilClasses
+      then
+        compilerError("cannot assign to a non-nilable owned variable from a nilable class");
```

Comment:
Should this error say "cannot copy-initialize a ... from a ..." here and elsewhere?

Reply:
If that felt too verbose, could also say "cannot *create* a ... from a ..."